pinot-spi/src/main/java/org/apache/pinot/spi/stream/RowMetadata.java

```diff
@@ -49,6 +49,17 @@ public interface RowMetadata {
    */
   long getRecordIngestionTimeMs();
 
+  /**
+   * Returns the creation timestamp associated with the record. In cases where the upstream ingestion pipeline is
+   * simple this timestamp matches the result of getRecordIngestionTimeMs();
+   *
+   * Expected to be used for stream-based sources.
+   *
+   * @return timestamp (epoch in milliseconds) when the row was initially created and ingested upstream for the first
+   *         time Long.MIN_VALUE if not available
+   */
+  long getRecordCreationTimeMs();
```

Comment:
Shouldn't we return the default value Long.MIN_VALUE to not break different streams' implementations?

Reply:
I was adding that in the override but yes we can do default instead

Reply:
addressed in commit 9bb0befe9a55c01bb3a5066a4b30137469819f38